### PR TITLE
Add source for LuaSnip.

### PR DESCRIPTION
**Pull request by L3MON4D3**
Hi, I've written a nvim-compe source for LuaSnip-snippets, is it alright if it's in nvim-compe or should I put it somewhere in LuaSnip?

### CODE REVIEW

File: lua/compe_luasnip/init.lua [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    dup = true;
+    menu = '[Snippets]';
+  }
+end
+
+function Source.determine(_, context)
+  if not luasnip_exists then
+    error("Cannot add source 'luasnip' without LuaSnip installed!")
+    return {}
+  end
```
Comment (hrsh7th) [inline lua/compe_luasnip/init.lua:22]: Please remove this error function.
If the user has a wrong configuration, it will be shown many many times.

Comment (L3MON4D3) [reply lua/compe_luasnip/init.lua:22]: Ah, nice.

File: lua/compe_luasnip/init.lua [hunk 2]
```diff
@@ -0,0 +1,60 @@
+local compe = require("compe")
+local Source = {}
+local luasnip_exists, luasnip = pcall(require, "luasnip")
```
Comment (hrsh7th) [inline lua/compe_luasnip/init.lua:3]: pcall can be removed.

**Comment by hrsh7th** [comment]
LGTM. Could you add `external source` section in doc/README.md?

File: lua/compe_luasnip/init.lua (hunk 2 above)
Comment (L3MON4D3) [reply lua/compe_luasnip/init.lua]: Right

**Comment by L3MON4D3** [comment]
Nice, is it fine under 'External-plugin'?

**Pull request merged**